src/movement.cpp

```diff
@@ -632,9 +634,9 @@ bool MoveEvent::configureEvent(const pugi::xml_node& node)
 	return true;
 }
 
-uint32_t MoveEvent::StepInField(Creature* creature, Item* item, const Position&)
+uint32_t MoveEvent::StepInField(std::shared_ptr<Creature> creature, std::shared_ptr<Item> item, const Position&)
 {
-	MagicField* field = item->getMagicField();
+	auto field = item->getMagicField();
```

Comment:
Could this be `const auto`
```suggestion
	const auto field = item->getMagicField();
```